arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/stream/concurrent/PubSub.kt

```diff
@@ -566,9 +565,9 @@ internal class DefaultPubSub<I, O, QS, S>(private val strategy: PubSub.Strategy<
   fun <X> update(f: (PubSub.PubSubState<I, O, QS, S>) -> Pair<PubSub.PubSubState<I, O, QS, S>, X>): X =
     state.modify { ps ->
       val (ps1, result) = f(ps)
-      val (ps2, _) = loop(ps1) { Unit }
-      Pair(ps2, result)
-    }
+      val (ps2, action) = loop(ps1) { Unit }
+      Pair(ps2, { action.invoke(); result })
```

Comment:
This was the bug fix, in case an outstanding offer was found. Its completion needs to be invoked.